tensorboard/webapp/feature_flag/views/feature_flag_page_component.ts

```diff
@@ -12,20 +12,52 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
-import {Component, Input} from '@angular/core';
-import {FeatureFlagState} from '../store/feature_flag_types';
+import {Component, EventEmitter, Input, Output} from '@angular/core';
+import {FeatureFlagType} from '../store/feature_flag_metadata';
+import {FeatureFlags} from '../types';
+import {FeatureFlagStatus, FeatureFlagStatusEvent} from './types';
+
 @Component({
   selector: 'feature-flag-page-component',
-  templateUrl: `feature_flag_page.ng.html`,
+  styleUrls: ['feature_flag_page_component.css'],
+  templateUrl: `feature_flag_page_component.ng.html`,
 })
 export class FeatureFlagPageComponent {
-  @Input() featureFlags!: FeatureFlagState;
+  @Input() featureFlagStatuses!: FeatureFlagStatus<keyof FeatureFlags>[];
+
+  @Output() flagChanged = new EventEmitter<FeatureFlagStatusEvent>();
+
+  @Output() allFlagsReset = new EventEmitter();
+
+  private getFormattedFlagValue(value: FeatureFlagType): string {
+    if (value === true) {
+      return 'Enabled';
+    }
+
+    if (value === false) {
+      return 'Disabled';
+    }
+
+    if (value === null || value === undefined) {
+      return 'null';
+    }
+
+    if (Array.isArray(value)) {
+      return JSON.stringify(value);
+    }
+
+    return value.toString();
+  }
 
-  getFlagKeys(): string[] {
-    return Object.keys(this.featureFlags);
+  isEditable(flagStatus: FeatureFlagStatus<keyof FeatureFlags>) {
+    return typeof flagStatus.defaultValue === 'boolean';
   }
 
-  flagChanged(flag: string) {
-    // TODO: dispatch action which stores new flag states and updates state.
+  formatFlagValue(value: FeatureFlagType): string {
```

Comment:
What is the logic in having formatFlagValue and getFormattedFlagValue as separate functions? Seems pretty straight forward to just have a single function.

Reply:
My logic was that `getFormattedFlag` essentially serializes the flag value while `formatFlagValue` decorates it. The naming does seem kinda confusing though so I'll rename them to hopefully make more sense.